silkrpc/commands/eth_api.cpp

```diff
@@ -148,11 +148,43 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_block_by_hash(const nlohman
     co_return;
 }
 
+asio::awaitable<void> EthereumRpcApi::handle_eth_get_block_transaction_count_by_hash(const nlohmann::json& request, nlohmann::json& reply) {
+    auto params = request["params"];
+    if (params.size() != 1) {
+        auto error_msg = "invalid handle_eth_get_block_transaction_count_by_hash params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+    auto block_hash = params[0].get<evmc::bytes32>();
+    SILKRPC_DEBUG << "block_hash: " << block_hash;
+
+    auto tx = co_await database_->begin();
+
+    try {
+        ethdb::kv::TransactionDatabase tx_database{*tx};
+
+        const auto block_with_hash = co_await core::rawdb::read_block_by_hash(tx_database, block_hash);
+        const auto tx = block_with_hash.block.transactions.size();
+
+        reply = make_json_content(request["id"], tx);
```

Comment:
The reply must be in `QUANTITY` format, i.e. you should use `"0x" + to_hex_no_leading_zeros(tx_count)` instead of `tx`

Reply:
ok